server-core/engine/src/main/java/io/infectnet/server/engine/core/world/CellularAutomaton.java

```diff
@@ -0,0 +1,372 @@
+ */
+class CellularAutomaton implements MapGeneratorStrategy{
+  /**
+   * The first type of cell, that the map consist of.
+    */
+  private static final boolean CAVE = true;
+
+  /**
+   * Sets how dense the initial grid is with living cells.
+   */
+  private static final float chanceToStartAlive = 0.4f;
+
+  /**
+   * The number of neighbours that cause an alive cell to die.
+   */
+  private static final int deathLimit = 4;
+
+  /**
+   * The number of neighbours that cause a dead cell to become alive.
+   */
+  private static final int birthLimit = 4;
+
+  /**
+   * The number of times we perform the simulation step.
+   */
+  private static final int numberOfSteps = 3;
+
+  /**
+   * The height of the map.
+   */
+  private final int height;
+
+  /**
+   * The width of the map.
+   */
+  private final int width;
+
+  /**
+   * The generated map.
+   */
+  private boolean[][] map;
+
+  /**
+   * The final product of the Automaton, with one cave system.
+   */
+  private boolean[][] finalMap;
+
+  /**
+   * Creates a Cellular Automaton with the limiting numbers.
+   * @param height the height of the map
+   * @param width the width of the map
+   */
+  CellularAutomaton(int height, int width) {
+    this.height = height;
+    this.width = width;
+  }
+
+  /**
+   * Generates a much simplified form of a {@link Map}, using booleans instead of {@link TileType}s.
+   * @return a boolean array containing the data about all tiles that were generated.
+   */
+  @Override
+  public boolean[][] generateMap(){
+    map = new boolean[height][width];
+
+    initializeMap();
+
+    for(int i = 0; i < numberOfSteps; ++i){
+      doSimulationStep();
+    }
+
+    map = eliminateIslands();
+
+    return map;
+  }
+
+  /**
+   * Initializes the array with true and false values placed at random.
+   */
+  private void initializeMap(){
+    for(int x = 0; x < width; ++x){
+      for(int y = 0; y < height; ++y){
+        if(random() < chanceToStartAlive){
+          map[y][x] = CAVE;
+        }
+      }
+    }
+  }
+
+  /**
+   * Performs a simulation step, in which it creates a new map using the values of the original,
+   * now old map and the pre-set values of deathLimit and birthLimit.
+   */
+  private void doSimulationStep(){
+    boolean[][] newMap = new boolean[height][width];
+
+    for(int y = 0; y < map.length; ++y){
+      for(int x = 0; x < map[0].length; ++x){
+        int aliveNeighbours = countAliveNeighbours(y, x);
+
+        if(map[y][x]){
+          newMap[y][x] = aliveNeighbours >= deathLimit;
+        }
+        else{
+          newMap[y][x] = aliveNeighbours > birthLimit;
+        }
+      }
+    }
+    map = newMap;
+  }
+
+  /**
+   * Counts the number of cells in a ring around (x,y) that are alive.
+   * @param x the first coordinate
+   * @param y the second coordinate
+   * @return the number of the alive cells found surrounding the (x,y) cell
+   */
+  private int countAliveNeighbours(int x, int y){
```

Comment:
The function has the following order of arguments: X, Y. However it is called in line 110 as

~~~~Java
int aliveNeighbours = countAliveNeighbours(y, x);
~~~~

Please handle consistently the coordinates and modify the code accordingly.